### PR TITLE
Services/biblio ref/process url pdf

### PR DESCRIPTION
**Pull request by leogail**

### CODE REVIEW

File: services/biblio-ref/v1/validate-pdf.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+for line in sys.stdin:
+    line0 = json.loads(line)
+    pdf_filename = line0["filename"]
+    
+    try:
+        references_structured = extract_references_from_file(pdf_filename)
+    except:
+        references_structured = [""]
```
Comment (parmentf) [inline services/biblio-ref/v1/validate-pdf.py:17]: **question:** ‏Why do you set one item in this list?

It should be the same with an empty list, shouldn't it?

File: services/biblio-ref/v1/validate-url.ini [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+mimeType = application/json
+
+post.operationId = post-v1-validate-url
+post.summary = Valide l'ensemble des références bibliographiques d'un PDF à partir de son URL
+post.description =  Extrait les références bibliographiques d'un PDF à partir de son URL puis utilise crossref pour:^M- valider les références bibliographiques^M- donner les DOIs s'ils existent^M- indiquer les éventuelles références rétractées.
+post.tags.0 = biblio-ref
+post.requestBody.content.application/json.schema.$ref = #/components/schemas/JSONStream
+post.requestBody.required = true
+post.responses.default.content.application/json.schema.$ref =  #/components/schemas/JSONStream
+post.responses.default.description = Le champ `value` est constitué d'un tableau constitués d'objets contenant les champs `doi`, `reference`, `url_pdf` et `status`.^M`status` prend plusieurs valeurs:^M- `found`si la référence est trouvée et valide,^M- `not_found` si elle est hallucinée ou non présente dans Crossref,^M- `retracted` si elle est rétractée. Indique les erreurs possibles `error_service` ou `error_data`.
```
Comment (parmentf) [inline services/biblio-ref/v1/validate-url.ini:11]: **typo:** ‏
```suggestion
post.responses.default.description = Le champ `value` est un tableau d'objets contenant les champs `doi`, `reference`, `url_pdf` et `status`.^M`status` prend plusieurs valeurs:^M- `found`si la référence est trouvée et valide,^M- `not_found` si elle est hallucinée ou non présente dans Crossref,^M- `retracted` si elle est rétractée. Indique les erreurs possibles `error_service` ou `error_data`.
```

File: services/biblio-ref/v1/validate-url.ini [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+post.parameters.0.description = Indenter le JSON résultant
+post.parameters.0.in = query
+post.parameters.0.name = indent
+post.parameters.0.schema.type = boolean
+
+post.requestBody.content.application/json.example.0.id = 1
+post.requestBody.content.application/json.example.0.value = https://cnrs.hal.science/hal-04461354/file/anki-gaillard-revol.pdf
+
```
Comment (parmentf) [inline services/biblio-ref/v1/validate-url.ini:19]: **todo:** ‏Add the example response.

File: services/biblio-ref/v1/validate-url.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+from bibref.bibref_functions import *
+import os
+from refextract import extract_references_from_file
+
+    
+for line in sys.stdin:
+    line0=json.loads(line)
+    # get URL of the PDF
+    url=line0['value']
+    name=str(round(random.random()*1000))+url.split('/')[-1]
```
Comment (parmentf) [inline services/biblio-ref/v1/validate-url.py:15]: **nitpick (non-blocking):** Will 0-999 filenames always be enough?

I'd prefer you to also use the current time to differentiate filenames: you don't know, one day there perhaps several thousands of queries will happen in a second ;)‏

File: services/biblio-ref/v1/validate.py [hunk 5]
```diff
@@ -9,56 +9,9 @@
...
-
-    # second case : no doi is found
-    else:
-        status,doi = verify_biblio(ref_biblio)
-        data["value"] = {"doi":doi, "status": status}
-        
-        json.dump(data, sys.stdout)
-        sys.stdout.write("\n")
+    
+    res = biblio_ref(ref_biblio)
```
Comment (parmentf) [inline services/biblio-ref/v1/validate.py:13]: **praise:** ‏Much simpler ! :+1:

**Review by parmentf — CHANGES_REQUESTED** [review]

File: services/biblio-ref/v1/validate-url.py (hunk 4 above)
Comment (leogail) [reply services/biblio-ref/v1/validate-url.py:15]: I hope so !

File: services/biblio-ref/v1/validate-pdf.py (hunk 1 above)
Comment (leogail) [reply services/biblio-ref/v1/validate-pdf.py:17]: not exactly : with an empty list, the result is value : [ ] but not  with a single empty string element, [""].

BUT an empty list is a better output i guess : i'll change it !

File: services/biblio-ref/examples.http [hunk 6]
```diff
@@ -38,3 +38,14 @@ Content-Type: application/json
...
+# @name v1validateUrl
+# Valide la référence bibliographique
+POST {{host}}/v1/validate-url?indent=true HTTP/1.1
+Content-Type: application/json
+
+[
+  {
+    "value":"https://cnrs.hal.science/hal-04461354/file/anki-gaillard-revol.pdf"
+  }
+]
```
[the remark this answers is not in the document]
Comment (leogail) [reply services/biblio-ref/examples.http:51]: Done ! But not sure if i did good with tests.hurl file. Could you check it @parmentf ?

[the remark this answers is not in the document]
Comment (leogail) [reply services/biblio-ref/examples.http:51]: nevermind ! file is ok this way

**Review by parmentf — CHANGES_REQUESTED** [review]

File: services/biblio-ref/tests.hurl [hunk 7]
```diff
@@ -76,4 +75,79 @@ HTTP 200
...
+Content-Type: application/json
+file,./example.pdf;
+
+
+HTTP 200
+# Capture the computing token
+[Captures]
+fr_computing_ref: jsonpath "$[0].value"
+[Asserts]
+variable "fr_computing_ref" exists
```
[the remark this answers is not in the document]
Comment (leogail) [reply services/biblio-ref/tests.hurl:153]: I understood what you are asking but i prefer do it with you for the first time

**Review by parmentf — APPROVED** [review]

**Pull request merged**